app/controllers/api/predictions_controller.rb

```diff
@@ -5,15 +5,32 @@ class PredictionsController < ApplicationController
 
     before_filter :authenticate_by_api_token
     before_filter :build_predictions, only: [:index]
+    before_filter :find_prediction, only: [:show, :update]
+
+    def create
+      @prediction = build_new_prediction
+
+      if @prediction.save
+        render json: @prediction
+      else
+        render json: @prediction.errors, status: :unprocessable_entity
+      end
+    end
 
     def index
       render json: @predictions
     end
 
-    def create
-      @prediction = build_new_prediction
+    def show
+      if authorized_to_view_prediction?
+        render json: @prediction
+      else
+        render json: unauthorized_user_message, status: :unauthorized
+      end
+    end
 
-      if @prediction.save
+    def update
+      if successfully_updated?
         render json: @prediction
       else
         render json: @prediction.errors, status: :unprocessable_entity
```

Comment:
Yeah, this method is trying to do too many things. I'll split it up with more appropriate names.